test/framework/src/main/java/org/elasticsearch/test/InternalTestCluster.java

```diff
@@ -898,17 +894,19 @@ public boolean isMasterEligible() {
         }
 
         Client client(Random random) {
-            if (closed.get()) {
-                throw new RuntimeException("already closed");
-            }
-            double nextDouble = random.nextDouble();
-            if (nextDouble < transportClientRatio) {
-                if (logger.isTraceEnabled()) {
-                    logger.trace("Using transport client for node [{}] sniff: [{}]", node.settings().get("node.name"), false);
+            synchronized (InternalTestCluster.this) {
```

Comment:
I think the synchronized block and closed check should be moved to getOrBuildNodeClient and getOrBuildTransportClient. That way it also covers nodeClient() and transportClient() methods. I think calling `InternalTestCluster.masterClient` is unsafe in current form.

Reply:
Done :)